### PR TITLE
Apply new dataset input system to sklearn tools

### PR DESCRIPTION
**Pull request by qiagu**
The tool took header by default was a trick. 😸

### CODE REVIEW

**Comment by qiagu** [comment]
@bgruening Could you please review this PR? There is one left, preprocess, yet I plan to put it in a separate PR.

File: tools/sklearn/nn_classifier.xml [hunk 1]
```diff
@@ -104,42 +107,60 @@ pickle.dump(classifier_object,open("$outfile_fit", 'w+'))
...
+            <param name="col2" value="5"/>
             <param name="selected_task" value="train"/>
             <param name="selected_algorithm" value="nneighbors"/>
             <param name="sampling_method" value="KNeighborsClassifier" />
             <param name="algorithm" value="brute" />
             <output name="outfile_fit" file="nn_model01.txt"/>
         </test>
         <test>
-            <param name="infile_train" value="train_set.tabular" ftype="tabular"/>
+            <param name="infile1" value="train_set.tabular" ftype="tabular"/>
```
Comment (bgruening) [inline tools/sklearn/nn_classifier.xml]: should this also be in a macro?

Comment (qiagu) [reply tools/sklearn/nn_classifier.xml]: Good point. You want me to change it?

Comment (bgruening) [reply tools/sklearn/nn_classifier.xml]: Not needed urgently or needed, but if we are at it we could change it at some point.

Comment (qiagu) [reply tools/sklearn/nn_classifier.xml]: Applying header/column selection to these train/predict tools doesn't seem to improve much, but looks bulky. Do you really like it?

Comment (bgruening) [reply tools/sklearn/nn_classifier.xml]: No ;) easier userinterface always wins.

Comment (qiagu) [reply tools/sklearn/nn_classifier.xml]: I also feel the gain/overhead ratio for this PR is low. Feel free to revoke the changes when they bother you.

**Pull request merged**